nntrainer/models/neuralnet.cpp

```diff
@@ -186,9 +188,22 @@ int NeuralNetwork::init() {
   int status = ML_ERROR_NONE;
   std::vector<TensorDim> previous_dim;
 
+  if (initialized) {
+    ml_loge("Error: Initializing the model again");
+    return ML_ERROR_NOT_SUPPORTED;
+  }
+
   status = isInitializable();
   NN_RETURN_STATUS();
 
+  /** Add loss layer if not already added */
+  if (layers.back()->getType() != LayerType::LAYER_LOSS) {
```

Comment:
How about add a warning if the user is overriding the loss set to the model?

Reply:
done.